pkg/middleware/json_payload_validator.go

```diff
@@ -0,0 +1,32 @@
+package middleware
+
+import(
+	"bytes"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"github.com/labstack/echo/v4"
+	"github.com/open-privacy/opv/pkg/apimodel"
+)
+
+func ValidateJSONPayload(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		var bodyBytes []byte
+		if c.Request().Body != nil {
+			bodyBytes, _ = ioutil.ReadAll(c.Request().Body)
+			// Restore the io.ReadCloser to its original state
+			c.Request().Body = ioutil.NopCloser(bytes.NewBuffer(bodyBytes))
+			// Use the content
+			if len(bodyBytes) > 0 {
+				jsonBody := make(map[string]interface{})
+				err := json.Unmarshal(bodyBytes, &jsonBody)
```

Comment:
you can also do `govalidator.IsJSON(string(bodyBytes))`, which is what we are using in other places.

https://github.com/asaskevich/govalidator/blob/7a23bdc65eef5f3783e782b436f3065eae3fc72d/validator.go#L439